presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -682,12 +694,53 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
             tableProperties.put(ORC_BLOOM_FILTER_FPP_KEY, String.valueOf(getOrcBloomFilterFpp(tableMetadata.getProperties())));
         }
 
+        // AVRO specific properties
+        String avroSchemaUrl = getAvroSchemaUrl(tableMetadata.getProperties());
+        if (avroSchemaUrl != null) {
+            HiveStorageFormat hiveStorageFormat = getHiveStorageFormat(tableMetadata.getProperties());
+            if (hiveStorageFormat != HiveStorageFormat.AVRO) {
+                throw new PrestoException(INVALID_TABLE_PROPERTY, format("Cannot specify %s table property for storage format: %s", AVRO_SCHEMA_URL, hiveStorageFormat));
+            }
+            tableProperties.put(AVRO_SCHEMA_URL_KEY, validateAndNormalizeAvroSchemaUrl(avroSchemaUrl, hdfsContext));
+        }
+
         // Table comment property
         tableMetadata.getComment().ifPresent(value -> tableProperties.put(TABLE_COMMENT, value));
 
         return tableProperties.build();
     }
 
+    private String validateAndNormalizeAvroSchemaUrl(String url, HdfsContext context)
+    {
+        try {
+            new URL(url).openStream().close();
+            return url;
+        }
+        catch (MalformedURLException e) {
+            // try locally
+            if (new File(url).exists()) {
+                // hive needs url to have a protocol
+                url = "file:///" + url;
+                return url;
+            }
+            else {
+                // try hdfs
+                try {
+                    if (!hdfsEnvironment.getFileSystem(context, new Path(url)).exists(new Path(url))) {
+                        throw new PrestoException(INVALID_TABLE_PROPERTY, "Cannot locate AVRO schema file " + url);
```

Comment:
Please add a colon to the end of the message, before the URL. We do this because it cleanly separates the user provided part.
```java
throw new PrestoException(INVALID_TABLE_PROPERTY, "Cannot locate AVRO schema file: " + url);
```
Same goes for the other errors.